apps/openmw/mwmechanics/actors.cpp

```diff
@@ -1591,6 +1596,30 @@ namespace MWMechanics
                 getActorsSidingWith(*it, out);
     }
 
+    void Actors::getActorsSidingWith(const MWWorld::Ptr &actor, std::set<MWWorld::Ptr>& out, std::map<const MWWorld::Ptr, const std::set<MWWorld::Ptr> >& cachedAllies) {
+        std::list<MWWorld::Ptr> followers = getActorsSidingWith(actor);
+
+        // If we have already found actor's allies, use the cache
+        std::map<const MWWorld::Ptr, const std::set<MWWorld::Ptr> >::const_iterator search = cachedAllies.find(actor);
+        if (search != cachedAllies.end())
+            out = search->second;
```

Comment:
Are you sure? Maybe I'm misunderstanding, but I think we don't want to append to out if we find it in the cache. If the actor is in the cache, we should be able to return its associated set. We want to append to out while doing the recursive function call, but I don't think it's possible to come across an actor already in the cache while we're doing that.

Sorry it's difficult to explain in words. Am I overlooking something?